samples: drivers: crypto: Align buffers

rettichschnidi opened the pull request:
This change completes what e1e19732bcdd00b55c5a7987580011eae0b640ee (samples: drivers: crypto: Aligned AES key) has started: For certain architectures (i.e. Si32), the alignment of the used buffers is important as the AES HW engine uses DMA and inherits its restrictions.

rettichschnidi committed:
samples: drivers: crypto: Align buffers

This change extends what e1e19732bcdd00b55c5a7987580011eae0b640ee
(samples: drivers: crypto: Aligned AES key) has started: For certain
architectures (i.e. Si32), the alignment of the used buffers is
important as the AES HW engine uses DMA and inherits its restrictions.

Signed-off-by: Reto Schneider <reto.schneider@husqvarnagroup.com>

rettichschnidi committed:
samples: drivers: crypto: Improve memory alignment

Commit e1e19732bcdd00b55c5a7987580011eae0b640ee (samples: drivers:
crypto: Aligned AES key) introduced the alignment for the key. According
to the commit message, the intention was to align the variable at 32 bit
boundaries. For this reason, passing 32 to __aligned seems like an
error, as its argument is denoted in byte and not bits.

Signed-off-by: Reto Schneider <reto.schneider@husqvarnagroup.com>

ceolin reviewed (changes requested):
Please, define a constant and use an integer directly, it is more readable and if we need to change its value in future (depending on the target) it will require less changes.

rettichschnidi committed:
samples: drivers: crypto: Make clang-format happy

Without this, check_compliance.py suggests to run clang-format on the
changed code lines.

Signed-off-by: Reto Schneider <reto.schneider@husqvarnagroup.com>

rettichschnidi commented:
> Please, define a constant and use an integer directly

Integer as opposed to some Kconfig symbol you mean? If so, I just did that.

rettichschnidi commented:
@ceolin Any chance this can be merged for Zephyr 4.0? Asking because my Si32 crypdriver (#..) needs this. Without I might have to/should/want to/? remove the Si32 driver until this PR makes it.

bjarki-andreasen commented on samples/drivers/crypto/src/main.c:
```diff
@@ -35,19 +35,19 @@ LOG_MODULE_REGISTER(main);
-	0x09, 0xcf, 0x4f, 0x3c
-};
+/* Some crypto drivers require IO buffers to be aligned, i.e. due to underlying DMA requirements. */
+#define IO_ALIGNMENT_BYTES 4
```
Since the key was aligned to 32 bytes, maybe that's a requirement for some platform? It may be safer just align all buffers to the largest alignment needed:
```
#define IO_ALIGNMENT_BYTES 32
```
which will also work for anything requiring 4 byte alignment (or 8, or 16) :)

rettichschnidi replied:
Reading the commit message of e1e19732bcdd00b55c5a7987580011eae0b640ee, it was always meant to be 4 bytes aligned.

Would rather get this in now, and see if anyone finds an issue when testing with RC2, instead of aligning by a (most likely) way too big value.

bjarki-andreasen replied:
ok, then 4 it shall be :)
